extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,145 +1,153 @@
+import java.net.URL;
+import java.util.List;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import org.json.JSONWriter;
-import com.google.refine.commands.Command;
-import org.deri.grefine.rdf.app.ApplicationContext;
-import com.google.refine.browsing.Engine;
-import com.google.refine.model.Project;
-import org.eclipse.rdf4j.rio.RDFFormat;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.FileItemFactory;
+import org.apache.commons.fileupload.FileUploadException;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
 import org.eclipse.rdf4j.rio.RDFParser;
+import org.eclipse.rdf4j.rio.jsonld.JSONLDParser;
 import org.eclipse.rdf4j.rio.nquads.NQuadsParser;
-import org.eclipse.rdf4j.rio.turtle.TurtleParser;
+import org.eclipse.rdf4j.rio.ntriples.NTriplesParser;
+import org.eclipse.rdf4j.rio.rdfjson.RDFJSONParser;
 import org.eclipse.rdf4j.rio.rdfxml.RDFXMLParser;
 import org.eclipse.rdf4j.rio.trig.TriGParser;
-import org.eclipse.rdf4j.rio.ntriples.NTriplesParser;
-import java.lang.System;
-import java.net.HttpURLConnection;
-import java.net.URL;
-import org.json.JSONException;
-import org.json.JSONObject;
-import org.eclipse.rdf4j.rio.Rio;
-import java.io.ByteArrayInputStream;
-import java.nio.charset.StandardCharsets;
-import org.apache.commons.fileupload.disk.DiskFileItemFactory;
-import org.apache.commons.fileupload.servlet.ServletFileUpload;
-import java.io.InputStream;
-import org.apache.commons.fileupload.FileItem;
-import org.apache.commons.fileupload.FileItemFactory;
-import java.util.List;
-import org.apache.commons.fileupload.FileUploadException;
-import java.io.InputStreamReader;
-import java.io.BufferedReader;
+import org.eclipse.rdf4j.rio.trix.TriXParser;
+import org.eclipse.rdf4j.rio.turtle.TurtleParser;
+import org.json.JSONWriter;
+import com.google.refine.commands.Command;
 
 /**
  * Enables automatic file format detection
+ * 
  * @author Shamanou van Leeuwen
  * @date May 24 2017
  *
  */
 
 
 /**
- * A command which detects the file format, should be configured in the controller 
- * and called from the javascript front-end.
+ * A command which detects the file format, should be configured in the controller and called from
+ * the javascript front-end.
  * 
  */
-public class DetectFileFormatCommand extends Command{
+public class DetectFileFormatCommand extends Command {
     RDFParser rdfParserttl = new TurtleParser();
     RDFParser rdfParserntriples = new NTriplesParser();
+    RDFParser rdfParserrdfxml = new RDFXMLParser();
+    RDFParser rdfParserJsonld = new JSONLDParser();
+    RDFParser rdfParserNQuads = new NQuadsParser();
+    RDFParser rdfParserJson = new RDFJSONParser();
+    RDFParser rdfParserTrix = new TriXParser();
+    RDFParser rdfParserTrig = new TriGParser();
 
     RDFParser[] parsers;
```

Comment:
1. This is missing an access modifier,
2. Can you explain why you chose an array and not a `Collection` type?

Reply:
we dont need a collection type, the number of parsers is predifined and doesnt change dynamically we also do not need keys, we do not need to know which parses is which because we can look at the values returned by the RDFParser parent class

Reply:
The choice for a collection is not only determined by the size. Ease of interaction, streams, etc.. A map is not a collection, so there are no keys involved.